woocommerce/woocommerce-admin
# Add navigation favorites data store

Fixes #6266 

Blocked by https://github.com/woocommerce/woocommerce-admin/issues/6136

This PR sets up the data store for favorites and all methods needed by the navigation to interact with favorites.

Note that the data store semantics and conventions are a slight departure from previous conventions used based on p2 discussions around data stores becoming more event focused.  Up for debate on the naming here, but I do find being "event-focused" to make a lot more sense.

### Detailed test instructions:

1.  Make sure tests pass.
2. Check follow-up PR that connects to the favorites UI. https://github.com/woocommerce/woocommerce-admin/pull/6287
3. Make sure new naming conventions make sense (discussion).
